include/emp/Evolve/Systematics.hpp

```diff
@@ -921,490 +1124,205 @@ namespace emp {
-    Ptr<taxon_t> GetMRCA() const;
-
-    /// Request the depth of the Most-Recent Common Ancestor; return -1 for none.
-    int GetMRCADepth() const;
-
-    /// Add information about a new organism, including its stored info and parent's taxon;
-    /// If you would like the systematics manager to track taxon age, you can also supply
-    /// the update at which the taxon is being added.
-    /// return a pointer for the associated taxon.
-    void AddOrg(ORG && org, WorldPosition pos, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG && org, WorldPosition pos, Ptr<taxon_t> parent=nullptr, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG && org, Ptr<taxon_t> parent=nullptr, int update=-1);
-
-    void AddOrg(ORG & org, WorldPosition pos, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG & org, WorldPosition pos, Ptr<taxon_t> parent=nullptr, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG & org, Ptr<taxon_t> parent=nullptr, int update=-1);
 
-
-    /// Remove an instance of an organism; track when it's gone.
-    bool RemoveOrg(WorldPosition pos, int time=-1);
-    bool RemoveOrg(Ptr<taxon_t> taxon, int time=-1);
-
-    void RemoveOrgAfterRepro(WorldPosition pos, int time=-1);
-    void RemoveOrgAfterRepro(Ptr<taxon_t> taxon, int time=-1);
-
-    /// Remove org from next population (for use with synchronous generations)
-    // bool RemoveNextOrg(WorldPosition pos, int time=-1);
-    // bool RemoveNextOrg(Ptr<taxon_t> taxon, int time=-1);
-
-    /// Climb up a lineage...
-    Ptr<taxon_t> Parent(Ptr<taxon_t> taxon) const;
+    // ===== Output functions ====
 
     /// Print details about the Systematics manager.
+    /// First prints setting, followed by all active, ancestor, and outside
+    /// taxa being stored. Format for taxa is
+    /// [ id | number of orgs in this taxon, number of offspring taxa of this taxon | parent taxon]
+    /// @param os output stream to print to
     void PrintStatus(std::ostream & os=std::cout) const;
 
-    /// Print whole lineage.
+    /// Print a whole lineage. Format: "Lineage:", followed by each taxon in the lineage, each on new line
+    /// @param taxon a pointer to the taxon to print the lineage of
+    /// @param os output stream to print to
     void PrintLineage(Ptr<taxon_t> taxon, std::ostream & os=std::cout) const;
 
+    /// Add a new snapshot function.
+    /// When a snapshot of the systematics is taken, in addition to the default
+    /// set of functions, all user-added snapshot functions are run. Functions
+    /// take a reference to a taxon as input and return the string to be dumped
+    /// in the file at the given key.
+    void AddSnapshotFun(const std::function<std::string(const taxon_t &)> & fun,
+                        const std::string & key, const std::string & desc="") {
+      user_snapshot_funs.emplace_back(fun, key, desc);
+    }
+
+    /// Take a snapshot of current state of taxon phylogeny.
+    /// WARNING: Current, this function assumes one parent taxon per-taxon.
```

Comment:
I'm not actually worried because there is no way to create a Taxon object with multiple parents. I think Alex just left that warning when he wrote the snapshot function. I think leaving it as a comment is a sufficient level of warning for future-proofing

Reply:
Ah, I was thinking along the lines of future proofing. That if we go in to add support for sexual lineages later the assert will trip if we forget to fix this.

Reply:
That's fair. Unfortunately there's not a good way to check for it without knowing what our implementation of sexual lineages would look like

Reply:
Ah, I see. The body is using `GetParent` which would break if it became something like `GetParents` or the return type changed to a list of parents etc.

A salient place to have a warning would probably be `LoadFromFile` function to ensure nobody's trying to load a sexual phylogeny. (whether that's a docstring comment or an assert)

Reply:
Totally agree! I added one of those in response to one of your earlier comment :)